tests/distributed/utils.py

```diff
@@ -2,30 +2,38 @@
 import socket
 import os
 
-def get_local_usable_addr():
-    """Get local usable IP and port
 
-    Returns
-    -------
-    str
-        IP address, e.g., '192.168.8.12:50051'
-    """
+def generate_ip_config(file_name, num_machines, num_servers):
+    """Get local IP and available ports, writes to file."""
+    # get available IP in localhost
     sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
     try:
         # doesn't even have to be reachable
         sock.connect(('10.255.255.255', 1))
-        ip_addr = sock.getsockname()[0]
+        ip = sock.getsockname()[0]
     except ValueError:
-        ip_addr = '127.0.0.1'
+        ip = '127.0.0.1'
     finally:
         sock.close()
+
+    # scan available PORT
+    ports = []
     sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    sock.bind(("", 0))
-    sock.listen(1)
-    port = sock.getsockname()[1]
```

Comment:
this will get a used port?

Reply:
this will get available/unused port. My change mainly aims to get a few continuous ports in case of `port already in use` when server gets started.